### PR TITLE
add tests for samcompare

### PR DESCRIPTION
**Pull request by yunfeiguo**
- [x] [RES-354](https://binatechnologies.atlassian.net/browse/RES-354)
- [x] Blocking PRs: ADD HERE IF ANY
* People
  - [x] Reviewers (>=1): @marghoob 
  - [x] Merger (>=1): @marghoob 
  - [x] FYI: ADD HERE IF ANY
- [x] Release note (if needed)
- [x] No code copied from outside Bina 
- [x] Tests (Unit, Workflow, ITL & Gherkin) are passing
- [x] PR labels, milestones & assignees

# Summary

* a functional test for short read sam compare
* fixed a bug of incorrectly throwing away path prefix for read map
* a functional test for long read sam compare
* a false positive test case for short read

### CODE REVIEW

**Comment by yunfeiguo** [comment]
@marghoob I'm done adding tests for SAMCompare. Could you please review the PR? Thanks!

File: src/main/java/com/bina/varsim/tools/evaluation/SAMcompare.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
     @Option(name = "-use_nonprimary", usage = "Do not skip non-primary alignments")
     boolean useNonPrimary = false;
     @Argument(usage = "One or more BAM files", metaVar = "bam_files ...", required = true)
-    private ArrayList<String> bam_filename = new ArrayList<>();
+    private List<String> bam_filename = new ArrayList<>();
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/SAMcompare.java]: 👍 for cleaning up stuff like this.

Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/SAMcompare.java]: `bam_filename` -> `bamFilename` to be consistent with the naming scheme.

**Comment by yunfeiguo** [comment]
Thank you!

**Pull request merged**